### ISSUE TITLE
MP#3 submission

### ISSUE THREAD

**Issue by Harry-2099** [Harry-2099/STA9750-2024-FALL#4]
Hi @michaelweylandt!
Heres  the link for the proj!
https://harry-2099.github.io/STA9750-2024-FALL/MP03.html

**Comment by michaelweylandt** [comment]
Thanks @Harry-2099 !

Peer Feedback:

- @ayrama
- @haolinnn
- @mussakone
- @ElissaLeung

It is now time for the peer feedback round for Mini-Project 03. Please
review @Harry-2099's submission for this mini-project and provide peer feedback.

Using the rubric at https://michael-weylandt.com/STA9750/miniprojects/mini03.html#rubric,
please grade their submission out of a total of 50 points. Note that this
rubric is slightly different than that used for Mini-Projects 01 and 02.

For each of the five categories, please give them a separate score and provide
a total (sum) score across the entire assignment. Feel free to assign extra
credit if you feel it is warranted (following the rubric).

If you give a score of less than 5 for any category, please provide a suggestion
for improvement. (You can also give suggestions for any element they did well -
more feedback is always great!)

As you go through this peer feedback exercise, think about what you particularly
like about this submission and how you can incorporate that approach in your future
work. If something is particularly insightful or creative, give some kudos!

Evaluators: This should take you around 15 minutes per peer feedback. You are
not required to engage in substantial back-and-forth with @Harry-2099, but you are of course
welcome to initiate a discussion.

@Harry-2099: please engage fully with your peers. They are here to help you!

Submission URL should be: https://Harry-2099.github.io/STA9750-2024-FALL/mp03.html

Feel free to link to other repos, the course documentation, or other useful
examples.

Thanks!
@michaelweylandt

CC: @charles-ramirez

**Comment by Harry-2099** [comment]
Hey guys, please use this link,

https://harry-2099.github.io/STA9750-2024-FALL/MP03.html

**Comment by ElissaLeung** [comment]
Hi @Harry-2099! 

I enjoyed reading through your submission.

Here's my feedback:

**Written Communication:** 9/10

I enjoyed reading through your project! Great use of transition sentences and context for the reader to understand the code being run throughout the project. I think you could have included some more analysis throughout the project beyond just explaining the code. For instance, a more in-depth comparison analysis of each ECV allocation method for task #7 could have helped the readers better understand the implication of each method on various elections beyond Gore vs. Bush.

**Project Skeleton:** 6/10

Almost all of the tasks were completed. I believe for task #4, we needed to adjust the code to unzip the downloaded zip shape files from Lewis et al. and the Census shape files we downloaded from tasks #1 and #2. Additionally, perhaps for task #7, you could have compared the outcomes of all the presidential elections beyond just the 2000 election with the different ECV allocation methods, for a more holistic analysis.

**Formatting & Display:** 10/10 + EC (Animated Plot)

Great job with the data tables and data visualizations in your project! Perhaps you can change the data table column names to clearer and cleaner names. I enjoyed the plethora of plots throughout the project. Also really liked the animated plot showing each election result. One minor suggestion is to include the ECV counts instead of the state abbreviations for tasks #5 and #6 (but this is personal preference). Other than some minor suggestions, I really enjoyed the overall formatting and display of the project.

**Code Quality:** 10/10

The code runs well. There are transition sentences surrounding code chunks to break down the code chunks and help the reader better understand the purpose of each chunk. One suggestion is to also include more comprehensive comments throughout each block so that these blocks are a little more digestible.

**Data Preparation:** 8/10

Good job with the data preparation at the beginning. You were able to automate the download process for the shape files from both Lewis et al. and the Census. My only comment is for task #4, I’m not sure if you wrote the function to automatically unzip and read in these shape files from the beginning.

**Total:** 43/50 + EC (animated plot)

Elissa

**Comment by ayrama** [comment]
Hey, Harry. Here is my feedback:

**Written Communication (9/10)**  
The report is well-organized and structured, with all tasks emphasized and bolded for clarity. The inclusion of folded code chunks and descriptive explanations of the plots and tables makes it reader-friendly. However, there could be more elaboration on the insights drawn from specific visualizations, such as the Congressional vs. Presidential candidate comparison.

**Project Skeleton (8/10)**  
The project follows the required structure, completing most tasks. The maps in tasks 5 and 6 suffer from overly saturated red and blue colors. Task 5's map is incomplete, lacking the number of electoral voters. While Task 6's animation is impressive, it omits Alaska and Hawaii, a detail that should have been included in both maps. Additionally, Task 7 lacks elaboration on all four voting schemes as outlined in the project instructions. While most tasks are addressed, these omissions prevent the skeleton from being fully complete.

**Formatting & Display (8/10)**  
The visualizations are mostly well-executed, with highlights such as the "Changes in Seats of the House of Representatives by State" plot, which stands out as excellent. However, some aspects could be improved, such as the confusing y-axis scale in the Congressional vs. Presidential candidate graph and the bright, eye-straining colors in Task 5's map. In the task 4: Fusion Fusses Table the table's column names could be replaced with more descriptive titles for clarity. In *Task 6: Animated Map of Party Changes* the animation works wellm thorough it excludes Alaska and Hawaii. 

**Code Quality (9/10)**  
The code is well-written and neatly organized, with folded chunks making the report tidy. However, additional comments explaining the logic behind certain steps would enhance readability and comprehension for external reviewers. Despite this, the code achieves the desired functionality effectively.

**Data Preparation (10/10)**  
The data preparation process is efficient and well-executed. Automation techniques are used appropriately, and the required datasets are handled systematically. This aspect of the project stands out as exemplary.



**Overall Score: 44/50**
The report demonstrates strong execution in most areas, particularly in data preparation and visualization. Minor improvements in formatting, completeness of tasks, and clarity of explanations would elevate the overall quality to an excellent standard.

**Comment by Harry-2099** [comment]
Hey @ayrama,
Thanks for the feedback
For task 5, when I view the plot on github Alaska and Hawaii are there, please double check if this is the case.

**Comment by haolinnn** [comment]
Hi @Harry-2099, good job on your project!

written communication: 9/10
It's very clear and flows, and you explain almost everything clearly. But I think you can include more explanations of the strategies that asked us to analyze in task 7.

project skeleton: 8/10
You complete all provided tasks throughout the project. But for task 5 and task 6, the maps are missing Hawaii.

Formatting & Display: 9 +1 extra credit for the animated map
The overall format and display are good, tables are well-organized. And I give extra credit for the animated map, you did it!

code quality: 9/10
The code is well-written and well-organized. For task 7, I understand you are not going into details of each part of the code because it is too long, but I think you can explain how you using code to get the results.

Data preparation: 10/10
Overall: 46/50

**Comment by mussakone** [comment]
Hi @Harry-2099 Here's what I think:

Written Communication: 9/10 
Your introduction was great and explanations for every part were great as well except for task 7 I feel like there could be much more of an explanation in this section.

project skeleton: 8/10
You complete all provided tasks throughout the project. Task 5 is not complete as it includes Alaska but no Hawaii.

Formatting and Display: 11/10
I thought your project flowed very nicely and had its fair share of maps, writing, and code but not too much of the three. In addition you get an additional point for your faceted/ animated map.

Code Quality: 9/10
Coding is clean and accurate I just wish since some code is massive you broke it up better or explained.

Data Preparation: 10/10
Data is automated

Overall: 47/50

**Comment by ayrama** [comment]
> Hey @ayrama, Thanks for the feedback For task 5, when I view the plot on github Alaska and Hawaii are there, please double check if this is the case.

You're right. Task 5 does include maps for Alaska and Hawaii, while Task 6 omits them. I've clarified the original feedback to reflect this.
